includes/compat/subscriptions/trait-wc-payment-gateway-wcpay-subscriptions.php

```diff
@@ -807,4 +810,113 @@ private function is_wcpay_subscription_renewal_order( WC_Order $renewal_order )
+		}
+
+		$result['setup_future_usage']                                = 'off_session';
+		$result['payment_method_options']['card']['mandate_options'] = [
+			'reference'       => $order->get_id(),
+			'amount'          => WC_Payments_Utils::prepare_amount( $subs_amount, $order->get_currency() ),
+			'amount_type'     => 'fixed',
+			'start_date'      => $subscription->get_time( 'date_created' ),
+			'interval'        => $subscription->get_billing_period(),
+			'interval_count'  => $subscription->get_billing_interval(),
+			'supported_types' => [ 'india' ],
+		];
+
+		// Multiple subscriptions per order needs:
+		// - Set amount type to maximum, to allow renews of any amount under the order total.
+		// - Set interval to sporadic, to not follow any specific interval.
+		// - Unset interval count, because it doesn't apply anymore.
+		if ( 1 < count( $subscriptions ) ) {
+			$result['payment_method_options']['card']['mandate_options']['amount_type'] = 'maximum';
+			$result['payment_method_options']['card']['mandate_options']['interval']    = 'sporadic';
+			unset( $result['payment_method_options']['card']['mandate_options']['interval_count'] );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Add an order note if the renew intent customer notification requires the merchant to authenticate the payment.
+	 * The note includes the charge attempt date and let the merchant know the need of an off-session step by the customer.
+	 *
+	 * @param WC_Order $order The renew order.
+	 * @param array    $processing Processing state from Stripe's intent response.
+	 * @return void
+	 */
+	public function maybe_add_customer_notification_note( WC_Order $order, array $processing = [] ) {
+		$approval_requested = $processing['card']['customer_notification']['approval_requested'] ?? false;
+		$completes_at       = $processing['card']['customer_notification']['completes_at'] ?? null;
+		if ( $approval_requested && $completes_at ) {
+			$attempt_date = wp_date( get_option( 'date_format', 'F j, Y' ), $completes_at, wp_timezone() );
+			$attempt_time = wp_date( get_option( 'time_format', 'g:i a' ), $completes_at, wp_timezone() );
+
+			$note = sprintf(
+			/* translators: 1) date in date_format or 'F j, Y'; 2) time in time_format or 'g:i a' */
+				__( 'The customer must authorize this payment via a notification sent to them by the bank which issued their card. The authorization must be completed before %1$s at %2$s, when the charge will be attempted.', 'woocommerce-payments' ),
+				$attempt_date,
+				$attempt_time
+			);
+
+			$order->add_order_note( $note );
+		}
+
+	}
+
+	/**
+	 * Get mandate ID parameter to renewal payment if exists.
+	 * Only required for subscriptions renewals for cards issued in India.
+	 * More details https://wp.me/pc4etw-ky
+	 *
+	 * @param WC_Order $renewal_order The subscription renewal order.
+	 * @return array Param to be included or empty array.
+	 */
+	public function get_mandate_param_for_renewal_order( WC_Order $renewal_order ): array {
```

Comment:
I wonder if it's necessary when we will only call `get_mandate_param_for_renewal_order()` within `scheduled_subscription_payment` 🤔 I assume it could only be called if Subscriptions are enabled.

In the other hand `get_mandate_params_for_order()` could be called for an order that's not a subscription.

WDYT?

Reply:
That makes sense! No need to check then.